lib/spack/spack/installer.py

```diff
@@ -1621,6 +1515,116 @@ def spec(self):
         return self.pkg.spec
 
 
+def build_process(pkg, build_kwargs):
+    """This implements the process forked for each build.
+
+    Has its own process and python module space set up by
+    build_environment.fork().
+
+    This function's return value is returned to the parent process.
+    """
+    kwargs = build_kwargs
+
+    keep_stage = kwargs.get('keep_stage', False)
+    install_source = kwargs.get('install_source', False)
+    skip_patch = kwargs.get('skip_patch', False)
+    verbose = kwargs.get('verbose', False)
+    fake = kwargs.get('fake', False)
+
```

Comment:
It is almost exactly the same. This change was made because pickling doesn't want to deal with functions defined inside of functions. The following sections are different:

```
     pkg_id = package_id(pkg)
     tty.debug('{0} Building {1} [{2}]'
               .format(pre, pkg_id, pkg.build_system_class))

     # get verbosity from do_install() parameter or saved value
     echo = verbose
     if spack.package.PackageBase._verbose is not None:
         echo = spack.package.PackageBase._verbose

     pkg.stage.keep = keep_stage

     # parent process already has a prefix write lock
     with pkg.stage:
```

and

```
 pid = '{0}: '.format(pkg.pid) if tty.show_pid() else ''
 pre = '{0}{1}:'.format(pid, pkg.name)
 pkg_id = package_id(pkg)

 tty.debug('{0} Building {1} [{2}]'
           .format(pre, pkg_id, pkg.build_system_class))

 # get verbosity from do_install() parameter or saved value
 echo = verbose
 if spack.package.PackageBase._verbose is not None:
     echo = spack.package.PackageBase._verbose

 pkg.stage.keep = keep_stage
 with pkg.stage:
```